src/modules/interfaces.ts

```diff
@@ -260,7 +260,13 @@ export class Mapper2<E> {
           .map(c => c.propertyName)
           .shift() ?? '';
       // Using + '' to coerce to string without worrying if `.toString()` exists, because JS
-      this.entityId = (e: E) => ((e as any)[cloudColumn] ?? (e as any)[primaryColumn]) + '';
+      this.entityId = (e: E) => {
+        if (cloudColumns && !(cloudColumns instanceof Error)) {
+          return cloudColumns.map(col => (e as any)[col]).join('|');
```

Comment:
can we assume the order will be always the same? or should we add some kind of sorting logic to the decorator?

Reply:
So the order of the array is determined by the order of the execution of the decorators, which is static and done only once, so yes, the order is the same every time we run here.

Even if the order is different between runs of the engine process itself, there would be no `memo` cache data to contradict it, and these `entityId`s are never stored in the database, so there is no vulnerability possible here.

Reply:
Granted, digging in here is a good idea, though, since there is *something* wrong in this PR that I haven't figured out, yet.